app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt

```diff
@@ -124,14 +124,13 @@ class LoginFragment : Fragment() {
 
         rootView.email.addTextChangedListener(object : TextWatcher {
             override fun afterTextChanged(s: Editable) {
-                findNavController(rootView).navigate(LoginFragmentDirections
-                    .actionLoginToAuthPop(redirectedFrom = safeArgs.redirectedFrom, email = s.toString()))
+                if (s.toString() != safeArgs.email)
+                    findNavController(rootView).navigate(LoginFragmentDirections
+                        .actionLoginToAuthPop(redirectedFrom = safeArgs.redirectedFrom, email = s.toString()))
```

Comment:
```java
 if (BuildConfig.FLAVOR == PLAY_STORE_BUILD_FLAVOR) {

            smartAuthViewModel.requestCredentials(SmartAuthUtil.getCredentialsClient(requireActivity()))

            smartAuthViewModel.id
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    email.text = SpannableStringBuilder(it)
                })

            smartAuthViewModel.password
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    password.text = SpannableStringBuilder(it)
                })

            smartAuthViewModel.apiExceptionCodePair.nonNull().observe(viewLifecycleOwner, Observer {
                SmartAuthUtil.handleResolvableApiException(
                    it.first, requireActivity(), it.second)
            })

            smartAuthViewModel.progress
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    progressDialog.show(it)
                })
        }
```
Here it fails.

Reply:
I think a workaround is to move this block of code 
```
 rootView.email.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable) {
 if (s.toString() != safeArgs.email)
                    findNavController(rootView).navigate(LoginFragmentDirections
                        .actionLoginToAuthPop(redirectedFrom = safeArgs.redirectedFrom, email = s.toString()))
   override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) { /*Do Nothing*/ }
            override fun onTextChanged(email: CharSequence, start: Int, before: Int, count: Int) { /*Do Nothing*/ }
        })
```
into here

```
if (safeArgs.email.isNotEmpty()) {
            rootView.email.text = SpannableStringBuilder(safeArgs.email)
        // HERE <------------------------
} else if (BuildConfig.FLAVOR == PLAY_STORE_BUILD_FLAVOR) {

            smartAuthViewModel.requestCredentials(SmartAuthUtil.getCredentialsClient(requireActivity()))

            smartAuthViewModel.id
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    email.text = SpannableStringBuilder(it)
                    Timber.d("DEBUGGING ${email.text.toString()}")
                })

            smartAuthViewModel.password
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    password.text = SpannableStringBuilder(it)
                })

            smartAuthViewModel.apiExceptionCodePair.nonNull().observe(viewLifecycleOwner, Observer {
                SmartAuthUtil.handleResolvableApiException(
                    it.first, requireActivity(), it.second)
            })

            smartAuthViewModel.progress
                .nonNull()
                .observe(viewLifecycleOwner, Observer {
                    progressDialog.show(it)
                })
        }
```

Reply:
Good idea. Do the changes and will test if both bugs will fix.

Reply:
I have pushed a new update, can you test it as I'm still struggling to set up the environment

Reply:
Sure